compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/optimizer/LargeMethodOptimizer.java

```diff
@@ -399,6 +430,38 @@ private void generateSplits(BIRFunction function, List<Split> possibleSplits,
         removeUnusedVarsAndSetVarUsage(function);
     }
 
+    private BIRBasicBlock handleNewFuncReturnVal(BIRFunction function, BIROperand splitFuncCallResultOp,
+                                                 BirScope lastInsScope, BIRBasicBlock currentBB, int newBBNum,
+                                                 List<BIRBasicBlock> newBBList, BIROperand splitLastInsLhsOp) {
+        BIROperand isErrorResultOp = generateTempLocalVariable(function, symbolTable.booleanType);
+        BIRNonTerminator errorTestIns = new BIRNonTerminator.TypeTest(null, symbolTable.errorType,
+                isErrorResultOp, splitFuncCallResultOp);
+        errorTestIns.scope = lastInsScope;
+        currentBB.instructions.add(errorTestIns);
+        BIRBasicBlock trueBB = new BIRBasicBlock(new Name("bb" + ++newBBNum));
+        BIRBasicBlock falseBB = new BIRBasicBlock(new Name("bb" + ++newBBNum));
+        currentBB.terminator = new BIRTerminator.Branch(null, isErrorResultOp, trueBB, falseBB);
+        currentBB.terminator.scope = lastInsScope;
```

Comment:
Shall we use the constructor which included the bir scope as a parameter? And consider all other places too.

Reply:
For `BIRTerminator` I will use such constructor. For `BIRNonTerminator`s there is no constructor which included the bir scope as a parameter.